tornado/test/autoreload_test.py

```diff
@@ -46,3 +47,71 @@ def test_reload_module(self):
             universal_newlines=True)
         out = p.communicate()[0]
         self.assertEqual(out, 'Starting\nStarting\n')
+
+    def test_reload_module_with_argv_preservation(self):
+        main = """\
+import os
+import sys
+from tornado import autoreload
+
+# This import will fail if path is not set up correctly
+import testapp
+
+print(autoreload._original_argv)
+sys.stdout.flush()
+if 'TESTAPP_STARTED' not in os.environ:
+    os.environ['TESTAPP_STARTED'] = '1'
+else:
+    # Force the autoreload to exit
+    autoreload.add_reload_hook(lambda: os._exit(0))
+    autoreload._reload()
+"""
+
+        touch = """\
```

Comment:
Why are you doing this through a separate process? This could just be done in the test process itself. Is it important that the reload happens because a file was touched? It's simpler to just call `_reload()` directly like the first test does (it's good to get more coverage of the underlying mechanism, but a little weird to tie it in to the test for argv preservation)